Core/src/org/sleuthkit/autopsy/filequery/FileSearch.java

```diff
@@ -239,6 +247,80 @@ static SearchResults runFileSearchDebug(String userName,
         return page;
     }
 
+    /**
+     * Get a summary for the specified AbstractFile if no TextSummarizers exist get
+     * the first bit of the file.
+     *
+     * @param file The AbstractFile to summarize.
+     *
+     * @return The summary or first bit of the specified file as a String.
+     */
+    @NbBundle.Messages({"FileSearch.documentSummary.noPreview=No preview available.",
+        "FileSearch.documentSummary.noBytes=No bytes read for document, unable to display preview."})
+    static String summarize(AbstractFile file) {
+        TextSummarizer localSummarizer = summarizerToUse;
+        if (localSummarizer == null) {
+            synchronized (searchCache) {
+                if (localSummarizer == null) {
+                    try {
+                        localSummarizer = getLocalSummarizer();
+                    } catch (IOException ignored) {
+                        //no summarizers being present is not unexpected
+                    }
+                }
+            }
+        }
+        if (localSummarizer == null) {
+            //no summarizer was found just grab the first bit of the file
+            return getFirstLines(file);
+        } else {
+            try {
+                //a summary of length 40 seems to fit without vertical scroll bars
+                return localSummarizer.summarize(file, 40);
+            } catch (IOException ex) {
+                return Bundle.FileSearch_documentSummary_noPreview();
+            }
+        }
+
+    }
+
+    /**
+     * Get the first bit of text from the specified AbstractFile.
+     *
+     * @param file The AbstractFile to get text from.
+     *
+     * @return The first bit of text from the specified AbstractFile.
+     */
+    private static String getFirstLines(AbstractFile file) {
+        try (Reader reader = TextExtractorFactory.getExtractor(file, null).getReader()) {
+            char[] cbuf = new char[PREVIEW_SIZE];
+            reader.read(cbuf, 0, PREVIEW_SIZE);
+            return Arrays.toString(cbuf);
+        } catch (IOException ex) {
+            return Bundle.FileSearch_documentSummary_noBytes();
+        } catch (TextExtractorFactory.NoTextExtractorFound | TextExtractor.InitReaderException ex) {
+            return Bundle.FileSearch_documentSummary_noPreview();
+        }
+    }
+
+    /**
+     * Get the first TextSummarizer found by a lookup of TextSummarizers.
+     *
+     * @return The first TextSummarizer found by a lookup of TextSummarizers.
+     *
+     * @throws IOException
+     */
+    private static TextSummarizer getLocalSummarizer() throws IOException {
+        Collection<? extends TextSummarizer> summarizers
+                = Lookup.getDefault().lookupAll(TextSummarizer.class
+                );
+        if (!summarizers.isEmpty()) {
+            summarizerToUse = summarizers.iterator().next();
+            return summarizerToUse;
+        }
+        throw new IOException("No summarizers found");
```

Comment:
Perhaps it would be better to return null instead of throwing an exception? Nothing has really gone wrong. And then you wouldn't need that empty catch block.